### PR TITLE
Converted output produced to character or numeric in correlation dialog

### LINKED ISSUE #7430: Converting data frame produced to character and numeric in correlation dialog
In correlation dialog:
When saving to a new data frame(**Display on data frame** control is checked) and **Display on Diagonal** control is *blank* ,the resulting matrix is currently character. Could therefore the columns have a (C) afterwards, to show this. Also, when the *NA* option is used, or a number, e.g. 1, then ideally the columns would be numeric

### PR DESCRIPTION
**Pull request by Vitalis95**
Fixes #7430 
@rdstern @lilyclements @Ivanluv 
This PR converts the output produced to either character or numeric depending on the value you input on `Display on Diagonal`  control.

### CODE REVIEW

File: instat/dlgCorrelation.vb [hunk 1]
```diff
@@ -96,11 +99,10 @@ Public Class dlgCorrelation
         ucrChkShave.SetValuesCheckedAndUnchecked("TRUE", "FALSE")
 
         ucrInputDisplayNas.SetParameter(New RParameter("na_print", 3))
-        dctNaPrint.Add("blank", Chr(34) & " " & Chr(34))
+        dctNaPrint.Add("blank", Chr(34) & "" & Chr(34))
```
Comment (N-thony) [inline instat/dlgCorrelation.vb:102]: dctNaPrint.Add("blank", Chr(34) & Chr(34))?

**Review by N-thony — CHANGES_REQUESTED** [review]
@Vitalis95 great, just a small comment.

**Comment by Vitalis95** [comment]
@N-thony its ok now.

**Review by N-thony — APPROVED** [review]
Great

File: instat/dlgCorrelation.vb [hunk 2]
```diff
@@ -284,13 +291,15 @@ Public Class dlgCorrelation
...
         clsFashionModelFunction.AddParameter("leading_zeros", "FALSE", iPosition:=2)
         clsFashionModelFunction.AddParameter("na_print", Chr(34) & " " & Chr(34), iPosition:=3)
 
         clsFashionDataFrameFunction.SetPackageName("corrr")
         clsFashionDataFrameFunction.SetRCommand("fashion")
         clsFashionDataFrameFunction.AddParameter("x", clsRFunctionParameter:=clsCorrelationFunction, iPosition:=0)
+        clsFashionDataFrameFunction.SetAssignTo("last_correlation")
         clsFashionDataFrameFunction.AddParameter("decimals", "2", iPosition:=1)
         clsFashionDataFrameFunction.AddParameter("leading_zeros", "FALSE", iPosition:=2)
         clsFashionDataFrameFunction.AddParameter("na_print", Chr(34) & " " & Chr(34), iPosition:=3)
```
Comment (shadrackkibet) [inline instat/dlgCorrelation.vb:305]: Why do you need the same R command repeated? This code is clearly not [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).

**Review by shadrackkibet — CHANGES_REQUESTED** [review]

File: instat/dlgCorrelation.vb [hunk 3]
```diff
@@ -306,6 +315,28 @@ Public Class dlgCorrelation
...
 
+        clsDataFrameFunction.SetRCommand("data.frame")
+
+        clsModelPipeOperator.SetOperation("%>%")
+        clsModelPipeOperator.AddParameter("left", clsRFunctionParameter:=clsDataFrameFunction, iPosition:=0)
+        clsModelPipeOperator.AddParameter("right", clsRFunctionParameter:=clsMutateFunction, iPosition:=1)
+
+        clsDataFramePipeOperator.SetOperation("%>%")
+        clsDataFramePipeOperator.AddParameter("left", clsRFunctionParameter:=clsDataFrameFunction, iPosition:=0)
+        clsDataFramePipeOperator.AddParameter("right", clsRFunctionParameter:=clsMutateFunction, iPosition:=1)
```
Comment (shadrackkibet) [inline instat/dlgCorrelation.vb:326]: Same case here! the left and right of this operator are the same!

**Review by shadrackkibet — CHANGES_REQUESTED** [review]
1) Empty space. Consider increasing the size of the dialog when the checkbox is checked and reducing otherwise.
image
2) Tab order

**Comment by Vitalis95** [comment]
@rdstern @shadrackkibet , I have made the changes with @Ivanluv

**Review by rdstern — CHANGES_REQUESTED** [review]
Excellent.  This seems to work well.
A very minor concern, is that, when expanded, the dialogue is quite long, and I worry slightly that this might be a problem on some machines with lower resolution.
I suggest moving the display options checkbox up a bit, and also reducing the distance down to the group-box labelled Display Options.

Then this is a good improvement to merge.

**Review by rdstern — APPROVED** [review]
Please can @shadrackkibet comment on this now and give the second approval.  I am approving, because the size of the dialogue is now ok.  The confusion on the data frame and the correlation object still exists, though we now correctly have all 3.  But the bottom checkbox says Name for Data Frame, and really means (also) the name of the correlation object.

**Pull request merged**